core/src/main/java/com/linecorp/armeria/client/endpoint/healthcheck/HealthCheckStrategy.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.client.endpoint.healthcheck;
+
+import java.util.List;
+
+import com.linecorp.armeria.client.Endpoint;
+import com.linecorp.armeria.client.endpoint.healthcheck.PartialHealthCheckStrategy.TargetCount;
+
+public interface HealthCheckStrategy {
```

Comment:
Given this is an advanced stuff and a user can break `HealthCheckedEndpointGroup` very easily by writing a broken implementation, how about making this interface and its implementation package-private, i.e. hide from the public API and adding `maxEndpointCount` and `maxEndpointRatio` to `AbstractHealthCheckedEndpointGroupBuilder`? Their default could be both `null` and they should be mutually exclusive (i.e. throw an `IllegalStateException` when a user tries to set both).

Reply:
I agree.  I also think it can be broken easily. I suggest we will discuss it when we find a new reason to make it public.